swarm_dial.go

```diff
@@ -35,18 +38,24 @@ var (
 
 	// ErrDialToSelf is returned if we attempt to dial our own peer
 	ErrDialToSelf = errors.New("dial to self attempted")
+
+	// ErrNoTransport is returned when we don't know a transport for the
+	// given multiaddr.
+	ErrNoTransport = errors.New("no transport for protocol")
 )
 
-// dialAttempts governs how many times a goroutine will try to dial a given peer.
+// DialAttempts governs how many times a goroutine will try to dial a given peer.
 // Note: this is down to one, as we have _too many dials_ atm. To add back in,
 // add loop back in Dial(.)
-const dialAttempts = 1
+const DialAttempts = 1
```

Comment:
Tests needed access to these and, really, there's no reason not to export them. They're currently constants but we can make them variables later if we want them to be configurable (or even make them swarm parameters).